typescript/src/game_loop/game-loop-timer.ts

```diff
@@ -1,106 +1,81 @@
+import { Observable, Subject } from "rxjs";
 
 type Callback = (delta: number) => void;
 
 interface StoppableTimer {
-    stop(): void;
+  stop(): void;
 }
 
 interface GameLoopTimer {
-    start(cb: Callback): void
+  start(cb: Callback): void;
 }
 
 interface GLRxJSTimer {
-    start(): Subject<number>;
+  start(): Subject<number>;
 }
 
-export default class GameLoopTimerImpl implements GameLoopTimer, StoppableTimer {
-    private tickRate: number;
-    private running: boolean;
-    private timerId?: ReturnType<typeof setTimeout>;
-
-    constructor(fps: number) {
-        this.tickRate = 1000 / fps;
-        this.running = false;
-    }
-
-    stop() {
-        this.running = false;
-        if (this.timerId) {
-            clearTimeout(this.timerId);
-            this.timerId = undefined;
-        }
-    }
-
-    start(cb: Callback): void {
-        this.running = true;
-        let lastTime: number = Date.now();
-        const run = () => {
-
-            if (!this.running) {
-                return;
-            }
-
-            const currentTime = Date.now();
-            const diff = currentTime - lastTime;
-
-            try {
-                cb(diff);
-            } catch (e) {
-                console.error(e);
-            }
-
-            const opDiff = Date.now() - currentTime;
-            lastTime = currentTime;
-
-            if (opDiff < this.tickRate) {
-                this.timerId = setTimeout(run, this.tickRate - opDiff);
-            } else {
-                setImmediate(run);
-            }
-        }
-
-        run();
+export default class GameLoopTimerImpl
+  implements GameLoopTimer, StoppableTimer
+{
+  private tickRate: number;
+  private running: boolean;
+  private timerId?: ReturnType<typeof setTimeout>;
+
+  constructor(fps: number) {
+    this.tickRate = 1000 / fps;
+    this.running = false;
+  }
+
+  stop() {
+    this.running = false;
+    if (this.timerId) {
+      clearTimeout(this.timerId);
+      this.timerId = undefined;
     }
-
+  }
+
+  start(cb: Callback): void {
+    this.running = true;
+    let lastTime: number = Date.now();
+    const run = () => {
+      if (!this.running) {
+        return;
+      }
+
+      const currentTime = Date.now();
+      const diff = currentTime - lastTime;
+
+      try {
+        cb(diff);
+      } catch (e) {
+        console.error(e);
+      }
+
+      const opDiff = Date.now() - currentTime;
+      lastTime = currentTime;
+
+      if (opDiff < this.tickRate) {
+        this.timerId = setTimeout(run, this.tickRate - opDiff);
+      } else {
+        setImmediate(run);
+      }
+    };
+
+    run();
+  }
 }
 
-export class GameLoopRxJS implements StoppableTimer, GLRxJSTimer {
-    private tickRate: number;
-    private ticker?: Subscription;
-    private subject: BehaviorSubject<number>;
-
-    constructor(fps: number) {
-        this.tickRate = 1000 / fps;
-        this.subject = new BehaviorSubject<number>(0);
```

Comment:
It turns out that this timer/interval was never ever shared with multiple observers, so the use of a `BehaviorSubject` was unnecessary.